Translation for `Strict Mode` (#871)
reactjs/ru.react.dev: src/content/reference/react/StrictMode.md

---

## Справочник {/*reference*/}

### `<StrictMode>` {/*strictmode*/}

> Используйте `StrictMode`, для активации дополнительных проверок и предупреждений для вложенного дерева компонентов:

Reviewer: ```suggestion
Используйте `StrictMode` для активации дополнительных проверок и предупреждений для вложенного дерева компонентов:
```

);
```

> [Больше примеров использования.](#usage)

Reviewer: ```suggestion
[Больше примеров использования ниже.](#usage)
```

> Строгий режим активирует следующие поведения в режиме разработки:

Reviewer: ```suggestion
Строгий режим активирует следующие варианты поведения в режиме разработки:
```

- Ваши компоненты будут [рендерится повторно](#fixing-bugs-found-by-double-rendering-in-development) чтобы можно было найти баги, вызванные нечистым рендерингом.

> - Ваши компоненты будут [повторно запускать эффекты](#fixing-bugs-found-by-re-running-effects-in-development), чтобы можно было найти баги возникающие из-за отсутствия сбрасывающей функции в эффекте.

Author: я смотрела, что там было:
| cleanup | сброс (*напр., сбрасываемый эффект, сбросить эффект, сбрасывающая функция*) |

поэтому пыталась немного заменить на синонимы)

- Ваши компоненты будут [проверяться на использование устаревших API.](#fixing-deprecation-warnings-enabled-by-strict-mode)

#### Пропсы {/*props*/}

`StrictMode` не принимает никаких пропсов.

#### Предостережения {/*caveats*/}

* Если вы используете `<StrictMode>`, то не сможете отключить его для части дерева. Это гарантирует, что все компоненты внутри `<StrictMode>` проходят проверки. Если две команды, работающие над продуктом, не могут прийти к соглашению, нужны ли им эти проверки, то они должны либо достичь компромисса, либо переместить `<StrictMode>` ниже по дереву. 

---

## Применение {/*usage*/}

### Как включить строгий режим для всего приложения {/*enabling-strict-mode-for-entire-app*/}

> Strict Mode включает дополнительные проверки для всего дерева компонентов внутри `<StrictMode>` в режиме разработки. Эти проверки позволяют находить распространенные баги в компонентах на ранней стадии разработки.

Reviewer 2: Пропустил в предыдущий раз "Strict Mode", поэтому предлагаю перевести как "строгий режим" раз уже везде перевели.
```suggestion
Строгий режим включает дополнительные проверки для всего дерева компонентов внутри `<StrictMode>` в режиме разработки. Эти проверки позволяют находить распространенные баги в компонентах на ранней стадии разработки.
```

Author: поправила

> Чтобы включить Strict Mode для всего приложения, оберните ваш корень вашего приложения в `<StrictMode>` при рендеринге:

Reviewer 2: Здесь аналогичная ситуация с "Strict Mode"
```suggestion
Чтобы включить строгий режим для всего приложения, оберните ваш корень вашего приложения в `<StrictMode>` при рендеринге:
```

Author: и тут поправила

Несмотря на то, что проверки строгого режима **выполняются только в режиме разработки,** они помогают вам находить ошибки, которые уже существуют в вашем коде, и их может быть сложно стабильно воспроизвести в продакшене. Строгий режим позволяет исправлять такие ошибки до того, как ваши пользователи сообщат о них.

<Note>

Строгий режим включает следующие дополнительные проверки в режиме разработки:

- Ваши компоненты будут [рендерится повторно](#fixing-bugs-found-by-double-rendering-in-development) чтобы можно было найти баги, вызванные нечистым рендерингом.
- Ваши компоненты будут [повторно запускать эффекты](#fixing-bugs-found-by-re-running-effects-in-development) чтобы можно было найти баги возникающие из-за отсутствия сбрасывающей функции в эффекте.
- Ваши компоненты будут [проверяться на использование устаревших APIs.](#fixing-deprecation-warnings-enabled-by-strict-mode)

**Все эти проверки работают только в режиме разработки и не оказывают никакого эффекта в продакшен-сборке.**

</Note>

---

### Как включить строгий режим для части приложения {/*enabling-strict-mode-for-a-part-of-the-app*/}

> Строгий режим может быть включён для любой части вашего приложения:

Author: поправила, но убрала "также", потому что кажется, оно тут не нужно

</Sandpack>

> Без строгого режима было легко пропустить ошибку, пока вы не добавили больше ре-рендеров. Строгий режим сразу показал ошибку. Он помогает вам находить баги перед тем, как вы отправите их вашей команде или пользователям.

Author: Заменила на:
которые могут испортить работу вашей команды и вызывать проблемы у ваших пользователей.

а то получается испортить работу пользователей - странно звучит

  }, [roomId]);
```

> Теперь, когда ваш эффект "сбрасывается" и удаляет устаревшие подключения, утечка устранена. Обратите внимание, что проблема не стала заметной, пока вы не добавили больше функциональности (выпадающий список).

Reviewer: ```suggestion
Теперь, когда ваш эффект "убирает" за собой устаревшие подключения, утечка устранена. Обратите внимание, что проблема не стала заметной, пока вы не добавили больше функциональности (выпадающий список).
```

</Sandpack>

Обратите внимание, как количество активных подключений в консоли больше не растет.

Без строгого режима было легко пропустить, что у вашего эффекта отсутствует сбрасывающая функция. Благодаря вызову *установки → сброса → установки* вместо *установки* для вашего эффекта в режиме разработки, строгий режим сделал отсутствие сбрасывающей функции более заметной.

> [Узнать больше о сбрасывающей функции в эффекте.](/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development)

Reviewer 2: ```suggestion
[Узнать больше о функции сброса эффекта.](/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development)
```

---

### Исправление предупреждений об устаревших методах в строгом режиме {/*fixing-deprecation-warnings-enabled-by-strict-mode*/}

> React предупреждает, если какой-либо компонент внутри дерева `<StrictMode>` использует какое-либо из следующих устаревших API:

Reviewer 2: ```suggestion
React предупреждает, если какой-либо компонент внутри дерева `<StrictMode>` использует одно из устаревших API:
```
Перефразировал

Author: да, так намного лучше, не заметила повтора

> * [`findDOMNode`](/reference/react-dom/findDOMNode). [Смотреть альтернативные решения.](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage)

Author: наверное если смотрите, то тогда надо везде добавить "тут", а то как-то не очень звучит.
Я бы оставила `смотреть`наверное
